src/test/java/uk/gov/justice/probation/courtcaseservice/controller/CourtCaseControllerPutByHearingIdIntTest.java

```diff
@@ -516,4 +521,39 @@ private List<PersonReferenceType> buildDefendantIdentifiers(String crn, String c
                 PersonReferenceType.builder().type("PNC").value(pnc).build()
         );
     }
+
+    @Test
+    void shouldPersistGivenPersonId_andCreateOneIfNotGiven_whenCreateOrUpdateHearingByHearingId() throws IOException {
+        //Person id given for defendant 1
+        String defendantId1 =   "1263de26-4a81-42d3-a798-bad802433318";
+        String personIdForDefendant1 = "45316811-6d65-4deb-a876-a6582e6566f7";
+
+        //No person id for defendant 2
+        String defendantId2 =   "6f014c2e-8be3-4a12-a551-8377bd31a7b8";
+
+        final var createHearingJason = FileUtils.readFileToString(caseDetailsExtendedUpdate, "UTF-8");
+
+        given()
+                .auth()
+                .oauth2(getToken())
+                .contentType(ContentType.JSON)
+                .accept(ContentType.JSON)
+                .body(createHearingJason)
+                .when()
+                .put(PUT_BY_HEARING_ID_ENDPOINT, JSON_HEARING_ID)
+                .then()
+                .statusCode(201);
+
+        // known person id
+        defendantRepository.findFirstByDefendantIdOrderByIdDesc(defendantId1)
+                .ifPresentOrElse(defendantEntity -> {
+                    assertThat(defendantEntity.getPersonId()).isEqualTo(personIdForDefendant1);
+                }, () -> fail("Person id not matching"));
+
+        // person id unknown so check if exist
+        defendantRepository.findFirstByDefendantIdOrderByIdDesc(defendantId2)
+                .ifPresentOrElse(defendantEntity -> {
+                    assertThat(defendantEntity.getPersonId()).isNotBlank();
```

Comment:
You could go further here and use a [regex](https://github.com/ministryofjustice/court-case-matcher/blob/f245d82c4d163c007adb96d9b486ba23d61736ee/src/test/java/uk/gov/justice/probation/courtcasematcher/TestUtils.java) to confirm it's a valid UUID.